### ISSUE TITLE
DenseNet.Train default Adam(lr=1e-3) overshoots memorization — MoreData_ShouldNotDegrade fails (lossLong > lossShort)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1393]
## Summary

\`DenseNetNetworkTests.MoreData_ShouldNotDegrade\` consistently fails on master: loss at 200 training iterations is HIGHER than at 50 iterations on the same (input, target) pair. The optimizer is diverging from the partially-memorized solution rather than converging to it.

## Repro

\`\`\`
$ dotnet test --filter \"FullyQualifiedName=AiDotNet.Tests.ModelFamilyTests.NeuralNetworks.DenseNetNetworkTests.MoreData_ShouldNotDegrade\" --framework net10.0
Failed!  - Failed: 1, Passed: 0
  Message: 200 iterations loss (0.208127) > 50 iterations loss (0.143550).
           Optimizer may be diverging with more training.
\`\`\`

(Numbers vary slightly run-to-run but lossLong is consistently \~50% higher than lossShort, well outside the 1e-4 \`MoreDataTolerance\`.)

## Root cause

DenseNet defaults to \`AdamOptimizer(InitialLearningRate=0.001)\` with no LR schedule and no weight decay (\`src/NeuralNetworks/DenseNetNetwork.cs\` line 131). On a single (image, target) memorization fixture:

1. By iter 50 the loss has dropped to \`\~0.15\` — the dense connectivity (concat of all previous layer outputs) gives every parameter many gradient paths, so Adam converges fast.
2. After that point the per-parameter gradient magnitudes drop toward zero but Adam's per-parameter \`v̂\` denominator also decays, so the effective update size doesn't shrink in lockstep. The optimizer overshoots and drifts away from the memorized point.
3. By iter 200 the loss has rebounded to \`\~0.21\`, ~45% worse than at iter 50.

Per Huang et al. 2017 (DenseNet paper §3 \"Training\"), the paper-faithful default is **SGD with momentum 0.9 + step-decay LR schedule** (initial 0.1, divide by 10 at 50% and 75% of training epochs), NOT raw Adam at 1e-3. Adam works on DenseNet but typically with a lower base LR (1e-4) or with a Cosine/StepLR schedule attached.

## Affected tests

- \`DenseNetNetworkTests.MoreData_ShouldNotDegrade\` (currently visible)
- \`DenseNetNetworkTests.TrainingError_ShouldNotExceedTestError\` (likely overlaps — same overshoot would inflate training MSE)

### Option A — paper-faithful SGD+momentum default

Change \`DenseNetNetwork<T>\`'s ctor default to:

\`\`\`csharp
_optimizer = optimizer ?? new SgdOptimizer<T, Tensor<T>, Tensor<T>>(this,
    new SgdOptimizerOptions { InitialLearningRate = 0.1, Momentum = 0.9 });
\`\`\`

Matches the Huang 2017 default. Existing users passing an explicit optimizer are unaffected.

### Option B — keep Adam but add a default LR decay

Wire a Cosine/StepLR scheduler into the default Adam config, decaying \`1e-3 → 1e-5\` over the user-specified epoch count. Less invasive but adds scheduler dependency.

### Option C — lower default Adam LR to 1e-4

Closest to the current behaviour, just slower convergence. Probably the smallest blast radius if production users are tuned to the current LR.

A is the principled fix. B is the modern-Adam idiom. C is the minimal change. Maintainer's call.

## NOT this issue

The Occupancy memorization invariant + the NEAT timeout are tracked separately ([PR #1391](https://github.com/ooples/AiDotNet/pull/1391) and [#1392](https://github.com/ooples/AiDotNet/issues/1392) respectively).
